File: 0_3.leetcode/java/Container_With_Most_Water.java
public class Container_With_Most_Water {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int answer = 0;

        while (left < right) {
            answer = Math.max(answer, Math.min(height[right], height[left]) * (right - left));

            if (height[right] < height[left]) {
                right--;
            } else {
                left++;
            }
        }

        return answer;
    }
}

File: 0_3.leetcode/kotlin/src/main/kotlin/Container_With_Most_Water.kt
fun main() {
    fun maxArea(height: IntArray): Int {
        var answer = 0
        var left = 0
        var right = height.size - 1

        while (left < right) {
            answer = maxOf(answer, minOf(height[left], height[right]) * (right - left))
            if (height[right] < height[left]) {
                right--
            } else {
                left++
            }
        }

        return answer;
    }

}